Modules/TPC/src/GenericHistogramCheck.cxx

```diff
@@ -195,36 +248,43 @@ void GenericHistogramCheck::beautify(std::shared_ptr<MonitorObject> mo, Quality
   double xText = 0;
   double yText = 0;
   double yText2 = 0;
-  auto h = dynamic_cast<TH1D*>((mo->getObject()));
+
   if (mHistDimension == 1) {
-    TLine* lineX = new TLine(mMeanX, h->GetMinimum() * 1.1, mMeanX, h->GetMaximum() * 1.1);
-    TLine* lineXEV = new TLine(mExpectedValueX, h->GetMinimum() * 1.1, mExpectedValueX, h->GetMaximum() * 1.1);
+
+    auto h1 = dynamic_cast<TH1F*>((mo->getObject()));
+    if (!h1) {
+      ILOG(Warning, Support) << "h1 not found in D= 1" << ENDM;
```

Comment:
I guess one should return here, otherwise it will crash accessing nullptr.
```suggestion
      ILOG(Warning, Support) << "h1 not found in D= 1" << ENDM;
      return;
```